pkg/util/yaml/decoder_test.go

```diff
@@ -125,6 +126,40 @@ stuff: 1
 	}
 }
 
+func TestDecodeBrokenYAML(t *testing.T) {
+	s := NewYAMLOrJSONDecoder(bytes.NewReader([]byte(`---
+	stuff:
+broken-indent: 1
+---       
+  `)), 100)
+	obj := generic{}
+	err := s.Decode(&obj)
+	if err == nil {
+		t.Fatal("expected error with json: prefix, got no error")
+	}
+	if !strings.HasPrefix(err.Error(), "yaml: line 1:") {
```

Comment:
@lavalamp actually I think the problem was tabs on the second line, I fixed this case... however yaml still reports line '2' where it should be '3' (I guess it does not count '---' as 'line'.